pkg/typesystem/optimizations.go

```diff
@@ -0,0 +1,101 @@
+package typesystem
+
+import (
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/openfga/language/pkg/go/graph"
+
+	"github.com/openfga/openfga/pkg/tuple"
+)
+
+// This file contains methods to detect whether an authorization model exhibits certain characteristics.
+// This information can then be used to increase performance of a Check request.
```

Comment:
This new file is a proposal - I can revert if we don't like it, or I can send a different PR with this update

Reply:
I can't say I agree with the naming but I could see the value of aggregating the methods that are used to identify a resolution strategy within its own file.

Reply:
I'm open to new names :)

Reply:
I love that we abstract the strategy here as I can envision list objects optimization will need something similar

Reply:
@yissellokta the idea is not to have JUST this method here, but all the other methods that try to assess whether an optimization is applicable.